app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -381,13 +381,7 @@ def get_content_view_environments(label = nil, organization = nil)
           where("#{Organization.table_name}.id = ?", organization.id)
       environments = environments.where("#{Katello::ContentViewEnvironment.table_name}.label = ?", label) if label
 
-      environments.delete_if do |env|
-        if env.content_view.default
-          !env.environment.readable?
-        else
-          !env.content_view.readable?
-        end
-      end
+      environments = environments.to_a.delete_if { |env| (env.content_view.default && !env.environment.readable?) || !env.content_view.readable? }
```

Comment:
@cpeters I am not seeing delete_if modify in place, are you seeing something different?

```
2.2.2 :047 > pools =Katello::Pool.where("subscription_id < 3")
+----+----------+----------+----------+----------+----------+----------+---------+----------+----------+----------+----------+-----+----------+----------+-----------+----------+
| id | cp_id    | creat... | updat... | subsc... | accou... | contr... | virtual | quantity | start... | pool_... | end_date | ram | multi... | consumed | virt_only | unmap... |
+----+----------+----------+----------+----------+----------+----------+---------+----------+----------+----------+----------+-----+----------+----------+-----------+----------+
| 8  | ff808... | 2015-... | 2015-... | 1        | 1212729  | 10219221 |         | 5        | 2013-... | NORMAL   | 2022-... |     |          | 0        | false     | false    |
| 9  | ff808... | 2015-... | 2015-... | 2        | 1212729  | 10169839 |         | 50       | 2013-... | NORMAL   | 2022-... |     |          | 0        | false     | false    |
| 10 | ff808... | 2015-... | 2015-... | 2        | 1212729  | 10169839 |         | -1       | 2013-... | UNMAP... | 2022-... |     |          | 0        | true      | true     |
+----+----------+----------+----------+----------+----------+----------+---------+----------+----------+----------+----------+-----+----------+----------+-----------+----------+
3 rows in set
2.2.2 :048 > pools.count
 => 3 
2.2.2 :050 > pools.to_a.delete_if { |x| x.subscription_id == 1 }
+----+----------+----------+----------+----------+----------+----------+---------+----------+----------+----------+----------+-----+----------+----------+-----------+----------+
| id | cp_id    | creat... | updat... | subsc... | accou... | contr... | virtual | quantity | start... | pool_... | end_date | ram | multi... | consumed | virt_only | unmap... |
+----+----------+----------+----------+----------+----------+----------+---------+----------+----------+----------+----------+-----+----------+----------+-----------+----------+
| 9  | ff808... | 2015-... | 2015-... | 2        | 1212729  | 10169839 |         | 50       | 2013-... | NORMAL   | 2022-... |     |          | 0        | false     | false    |
| 10 | ff808... | 2015-... | 2015-... | 2        | 1212729  | 10169839 |         | -1       | 2013-... | UNMAP... | 2022-... |     |          | 0        | true      | true     |
+----+----------+----------+----------+----------+----------+----------+---------+----------+----------+----------+----------+-----+----------+----------+-----------+----------+
2 rows in set
2.2.2 :051 > pools.count
 => 3 
2.2.2 :052 > pools = pools.to_a.delete_if { |x| x.subscription_id == 1 }
+----+----------+----------+----------+----------+----------+----------+---------+----------+----------+----------+----------+-----+----------+----------+-----------+----------+
| id | cp_id    | creat... | updat... | subsc... | accou... | contr... | virtual | quantity | start... | pool_... | end_date | ram | multi... | consumed | virt_only | unmap... |
+----+----------+----------+----------+----------+----------+----------+---------+----------+----------+----------+----------+-----+----------+----------+-----------+----------+
| 9  | ff808... | 2015-... | 2015-... | 2        | 1212729  | 10169839 |         | 50       | 2013-... | NORMAL   | 2022-... |     |          | 0        | false     | false    |
| 10 | ff808... | 2015-... | 2015-... | 2        | 1212729  | 10169839 |         | -1       | 2013-... | UNMAP... | 2022-... |     |          | 0        | true      | true     |
+----+----------+----------+----------+----------+----------+----------+---------+----------+----------+----------+----------+-----+----------+----------+-----------+----------+
2 rows in set
2.2.2 :053 > pools.count
 => 2 
```